x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -270,6 +277,53 @@ export class TelemetryReceiver implements ITelemetryReceiver {
     return this.esClient.search(query, { meta: true });
   }
 
+  public async fetchEndpointMetadata(executeFrom: string, executeTo: string) {
+    if (this.esClient === undefined || this.esClient === null) {
+      throw Error('elasticsearch client is unavailable: cannot retrieve elastic endpoint metrics');
+    }
+
+    const query: SearchRequest = {
+      expand_wildcards: ['open' as const, 'hidden' as const],
+      index: `.ds-metrics-endpoint.metadata-*`,
+      ignore_unavailable: false,
+      size: 0, // no query results required - only aggregation quantity
+      body: {
+        query: {
+          range: {
+            '@timestamp': {
+              gte: executeFrom,
+              lt: executeTo,
+            },
+          },
+        },
+        aggs: {
+          endpoint_metadata: {
+            terms: {
+              field: 'agent.id',
+              size: this.max_records,
```

Comment:
Looks good. We might port this to a PIT aggregation in the future if we find this to be a bottleneck - but I'm sure it is fine.